python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -579,6 +583,92 @@ async def dome_el_in_position(self) -> str:
         await self._dome_el_in_position.wait()
         return "Dome elevation in position."
 
+    async def wait_for_dome_state(
+        self,
+        expected_states: set[MTDome.MotionState],
+        bad_states: set[MTDome.MotionState],
+        timeout: float,
+        check_in_position: bool = False,
+    ) -> None:
+        """Wait for a specific dome state.
+
+        Parameters
+        ----------
+        expected_states : set[MTDome.MotionState]
+            Valid states to transition into while un-parking.
+        bad_states : set[MTDome.MotionState]
+            States that are not allowed while un-parking and should raise an
+            error.
+        timeout : float
+            Maximum time to wait for the correct state.
+
+        Raises
+        ------
+        RuntimeError
+            If a bad state is encountered or the expected state is not reached
+             in time.
+        """
+
+        # Helper function checks if dome's ready.
```

Comment:
I am not sure how useful this comment is. I think the method name is descriptive enough that it is clear what this method is for. If you really want to make a comment, make it a docstring.

Reply:
Removed, and sorry for the extra comments, I'll be more sparse going forward.